config/jobs/kubernetes/publishing-bot/publishing-bot-periodics.yml

```diff
@@ -0,0 +1,87 @@
+periodics:
+- name: periodic-publishing-bot
+  cluster: k8s-infra-prow-build-trusted
+  annotations:
+    testgrid-dashboards: sig-release-publishing-bot
+    testgrid-tab-name: periodic
+  spec:
+    initContainers:
+    - name: initialize-repos
+      command:
+      - /init-repo
+      - --alsologtostderr
+      - --config=/etc/munge-config/config
+      - --rules-file=/etc/publisher-rules/config
+      - 2>&1
+      image: gcr.io/k8s-staging-publishing-bot/k8s-publishing-bot
+      imagePullPolicy: Always
+      resources:
+        requests:
+          cpu: 300m
+          memory: 200Mi
+        limits:
+          cpu: 2
+          memory: 1639Mi
+      volumeMounts:
+      - mountPath: /etc/munge-config
+        name: munge-config
+      - mountPath: /go-workspace
+        name: publisher-gopath
+      - mountPath: /etc/publisher-rules
+        name: publisher-rules
+      - mountPath: /.cache
+        name: cache
+    containers:
+    - name: publisher
+      command:
+      - /publishing-bot
+      - --alsologtostderr
+      - --config=/etc/munge-config/config
+      - --token-file=/etc/secret-volume/token
+      - --interval=0
+      - --server-port=8080
+      - 2>&1
+      image: gcr.io/k8s-staging-publishing-bot/k8s-publishing-bot
+      imagePullPolicy: Always
+      resources:
+        requests:
+          cpu: 300m
+          memory: 200Mi
+        limits:
+          cpu: 2
+          memory: 1639Mi
+      volumeMounts:
+      - mountPath: /etc/munge-config
+        name: munge-config
+      - mountPath: /etc/publisher-rules
+        name: publisher-rules
+      - mountPath: /etc/secret-volume
+        name: secret-volume
+      - mountPath: /netrc
+        name: netrc
+      - mountPath: /gitrepos
+        name: repo
+      - mountPath: /go-workspace
+        name: publisher-gopath
+      - mountPath: /.cache
+        name: cache
+    volumes:
+    - name: munge-config
+      configMap:
+        name: publisher-config
+    - name: publisher-rules
+      configMap:
+        name: publisher-rules
```

Comment:
These 2 configmaps also need to be created and available in the namespace, so the bot can use them

Reply:
Thinking out loud. If we should have a command added in init container that fetches the latest config and uses it, instead of relying via the configmap. This way we can make sure that , even updates to configs that we create in repo are available to the bot.

Otherwise we will need the configmap to be created in the cluster and updated manually every time a change happens. Again a change to the config is very very rare.